libcfnet/classic.c

```diff
@@ -100,12 +101,17 @@ int RecvSocketStream(int sd, char buffer[CF_BUFSIZE], int toget)
                 return -1;
             }
         }
-        else if (got == 0)                /* peer has closed the connection */
+        else if (got == 0)          /* peer has closed the connection early */
         {
-            break;
+            Log(LOG_LEVEL_ERR,
+                "Peer closed connection early (received=%dB, expecting=%dB)",
+                already, toget);
+            buffer[already] = '\0';
+            /* TODO Why return 0? Early close is an error so -1 is better. */
+            return 0;
         }
     }
-    assert(already <= toget);
+    assert(already == toget);
```

Comment:
Maybe it would be safer here to add some condition like if (already!=toget) {} to make sure that this is checked in release not only debug.